sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -487,6 +487,88 @@ case class GpuInitCap(child: Expression) extends GpuUnaryExpression with Implici
+  override def right: Expression = repeatTimes
+  override def dataType: DataType = input.dataType
+  override def inputTypes: Seq[AbstractDataType] = Seq(StringType, IntegerType)
+
+  def doColumnar(input: GpuScalar, repeatTimes: GpuColumnVector): ColumnVector = {
+    assert(input.dataType == StringType)
+
+    withResource(GpuColumnVector.from(input, repeatTimes.getRowCount.asInstanceOf[Int],
+                 input.dataType)) {
+      replicatedInput => doColumnar(replicatedInput, repeatTimes)
+    }
+  }
+
+  def doColumnar(input: GpuColumnVector, repeatTimes: GpuColumnVector): ColumnVector = {
+    val repeatTimesCV = repeatTimes.getBase
+
+    // Compute the output size to check for overflow.
+    withResource(input.getBase.repeatStringsSizes(repeatTimesCV)) { outputSizes =>
+      if (outputSizes.getTotalSize > Int.MaxValue.asInstanceOf[Long]) {
+        throw new RuntimeException("Output strings have total size exceed maximum allowed size")
+      }
+
+      // Finally repeat the strings using the pre-computed strings' sizes.
+      input.getBase.repeatStrings(repeatTimesCV, outputSizes.getStringSizes)
+    }
+  }
+
+  def doColumnar(input: GpuColumnVector, repeatTimes: GpuScalar): ColumnVector = {
+    assert(repeatTimes.dataType == IntegerType)
+
+    if (!repeatTimes.isValid) {
+      // If the input scala repeatTimes is invalid, the results should be all nulls.
+      // This is equivalent to having repeatTimes == 0.
+      input.getBase.repeatStrings(0)
+    } else {
+      repeatTimes.getValue match {
+        case repeatTimesVal: Int => { // only support Int type for this overload
+          // Get the input size to check for overflow for the output.
+          // Note that this is not an accurate check since the total buffer size of the input
+          // strings column may be larger than the total length of strings that will be repeated in
+          // this function.
+          val inputBufferSize = input.getBase.getData.getLength
+          if (repeatTimesVal > 0 && inputBufferSize > Int.MaxValue / repeatTimesVal) {
+            throw new RuntimeException("Output strings have total size exceed maximum allowed size")
+          }
+
+          // Finally repeat the strings.
+          input.getBase.repeatStrings(repeatTimesVal)
+        }
+        case _ => throw new IllegalStateException("Invalid data type for repeatTimes " +
+            "(must be INT32)")
+      }
+    }
+  }
+
+  def doColumnar(numRows: Int, input: GpuScalar, repeatTimes: GpuScalar): ColumnVector = {
+    assert(input.dataType == StringType && repeatTimes.dataType == IntegerType)
+
+    if (!repeatTimes.isValid) {
+      // If the input scala repeatTimes is invalid, the results should be all nulls.
+      // This is equivalent to having repeatTimes == 0.
+      withResource(input.getBase.repeatString(0)) {
+        repeatedString => ColumnVector.fromScalar(repeatedString, 1)
+      }
+    } else {
+      repeatTimes.getValue match {
```

Comment:
The code above is already asserting on the datatype, so the match can be replaced with `repeatTimes.getBase.getInt`.